ledger-core-cosmos/src/cosmos/database/CosmosLikeTransactionDatabaseHelper.cpp

```diff
@@ -75,85 +75,102 @@ namespace ledger {
+                auto &content = boost::get<cosmos::MsgUndelegate>(msg.content);
+                content.delegatorAddress = row.get<std::string>(COL_DELADDR);
+                content.validatorAddress = row.get<std::string>(COL_VALADDR);
+                soci::stringToCoin(row.get<std::string>(COL_AMOUNT), content.amount);
+            } break;
+            case api::CosmosLikeMsgType::MSGREDELEGATE: {
+                msg.content = cosmos::MsgRedelegate();
+                auto &content = boost::get<cosmos::MsgRedelegate>(msg.content);
+                content.delegatorAddress = row.get<std::string>(COL_DELADDR);
+                content.validatorSourceAddress = row.get<std::string>(COL_VALSRCADDR);
+                content.validatorDestinationAddress = row.get<std::string>(COL_VALDESTADDR);
+                soci::stringToCoin(row.get<std::string>(COL_AMOUNT), content.amount);
+            } break;
+            case api::CosmosLikeMsgType::MSGSUBMITPROPOSAL: {
+                msg.content = cosmos::MsgSubmitProposal();
+                auto &content = boost::get<cosmos::MsgSubmitProposal>(msg.content);
+                content.content.type = row.get<std::string>(COL_PROPTYPE);
+                content.content.title = row.get<std::string>(COL_PROPTITLE);
+                content.content.description = row.get<std::string>(COL_PROPDESC);
+                content.proposer = row.get<std::string>(COL_PROPOSER);
+                soci::stringToCoins(row.get<std::string>(COL_AMOUNT), content.initialDeposit);
+            } break;
+            case api::CosmosLikeMsgType::MSGVOTE: {
+                msg.content = cosmos::MsgVote();
+                auto &content = boost::get<cosmos::MsgVote>(msg.content);
+                content.voter = row.get<std::string>(COL_VOTER);
+                content.proposalId = row.get<std::string>(COL_PROPID);
+                content.option =
+                    api::from_string<api::CosmosLikeVoteOption>(row.get<std::string>(COL_VOTEOPT));
+            } break;
+            case api::CosmosLikeMsgType::MSGDEPOSIT: {
+                msg.content = cosmos::MsgDeposit();
+                auto &content = boost::get<cosmos::MsgDeposit>(msg.content);
+                content.depositor = row.get<std::string>(COL_DEPOSITOR);
+                content.proposalId = row.get<std::string>(COL_PROPID);
+                soci::stringToCoins(row.get<std::string>(COL_AMOUNT), content.amount);
+            } break;
+            case api::CosmosLikeMsgType::MSGWITHDRAWDELEGATIONREWARD: {
+                msg.content = cosmos::MsgWithdrawDelegationReward();
+                auto &content = boost::get<cosmos::MsgWithdrawDelegationReward>(msg.content);
+                content.delegatorAddress = row.get<std::string>(COL_DELADDR);
+                content.validatorAddress = row.get<std::string>(COL_VALADDR);
+            } break;
+            case api::CosmosLikeMsgType::MSGMULTISEND: {
+                msg.content = cosmos::MsgMultiSend();
```

Comment:
Add // TODO ?

Reply:
We'll see the empty message on the live side if necessary. I don't expect the history to need to handle these messages yet

Reply:
But I think I actually changed that in another PR

Reply:
I did not, this is a new ticket